src/Style.hs

```diff
@@ -300,130 +314,145 @@ initSpec = StySpec { spType = C.PointT,
                      spShpMap = M.empty
                    }
 
--- | 'getDictAndFns' is the top-level function used by "Runtime", which returns a dictionary of Style configuration and all objective and constraint functions generated by Style
+-- | 'getDictAndFns' is the top-level function used by "Runtime", which returns a dictionary of Style objects and all objective and constraint functions generated by Style
 -- TODO: maybe generate objects directly?
 getDictAndFns :: (Autofloat a) =>
-    ([C.SubDecl], [C.SubConstr]) -> StyProg
-    -> (StyDict, [(ObjFnOn a, Weight a, [Name], [a])], [(ConstrFnOn a, Weight a, [Name], [a])])
-getDictAndFns (decls, constrs) blocks = foldl procBlock (initDict, [], []) blocks
+    ([C.SubDecl], [C.SubConstr]) -> StyProg -> StyContext a
+getDictAndFns (decls, constrs) blocks =
+    foldl procBlock (initDict, [], []) blocks
     where
-        res :: [(C.SubType, String, [String])]
-        res = getSubTuples decls ++ getConstrTuples constrs
-
-        ids :: [(C.SubType, String)]
-        ids = map (\(x, y, z) -> (x, y)) res
-
-        initDict :: M.Map String StySpec
+        -- idsWithArgs :: [(C.SubType, String, [String])]
+        idsWithArgs = getSubTuples decls ++ getConstrTuples constrs
+        -- initDict :: M.Map String StySpec
         initDict = foldl (\m (t, n, a) ->
-                     M.insert n (initSpec { spId = n, spType = t, spArgs = a }) m) M.empty res
+                     M.insert n (initSpec { spId = n, spType = t, spArgs = a }) m) M.empty idsWithArgs
 
--- Helper functions for procBlock
-
--- removeDups :: [[(VarMap, StySpec)]]
--- TODO: really a helper function. consider moving to "Utils"
-cartesianProduct :: [[a]] -> [[a]]
-cartesianProduct = foldr f [[]] where f l a = [ x:xs | x <- l, xs <- a ]
+--------------------------------------------------------------------------------
+-- Selector implementation
+-- GOAL: given a list of selectors and all Substance identifiers, output a
```

Comment:
can you give an example here of a "list of selectors"? and what are the semantics of a list of selectors? is it AND (match all of them) or OR (match at least one of them)?

Reply:
I did try to cover it the [Style document](https://github.com/wodeni/notes-public/blob/master/penrose/style-compiler-doc.md#pattern-matching). I'll add examples to the doc for better explanation.